base_placement_plugin/src/place_base.cpp

```diff
@@ -110,7 +182,184 @@ void PlaceBase::setBasePlaceParams(int base_loc_size, int high_score_sp)
   }
 }
 
-void PlaceBase::findbase(std::vector< geometry_msgs::Pose > grasp_poses)
+//Creating <spheres, ri> and <highscoringsp> multimap from the <sphere, poses> the bool determines wherther to make it 2d/3d
+void PlaceBase::createSpheres(std::multimap< std::vector< double >, std::vector< double > > basePoses,
+                   std::map< std::vector< double >, double >& spColor, std::vector< std::vector< double > >& highScoredSp, bool reduce_D)
+{
+  kinematics::Kinematics k;
+  std::vector<int> poseCount;
+  poseCount.reserve(basePoses.size());
+  for (std::multimap< std::vector< double >, std::vector< double > >::iterator it = basePoses.begin(); it != basePoses.end();++it)
+  {
+    int num = basePoses.count(it->first);
+    poseCount.push_back(num);
+  }
+  std::vector<int>::const_iterator it;
+  it = max_element(poseCount.begin(), poseCount.end());
+  int max_number = *it;
+  it = min_element(poseCount.begin(), poseCount.end());
+  int min_number = *it;
+  if(reduce_D)
+  {
+    for (std::multimap< std::vector< double >, std::vector< double > >::iterator it = basePoses.begin(); it != basePoses.end();++it)
+    {
+      if((it->first)[2] < 0.051 && (it->first)[2] > -0.051)
```

Comment:
What is the 0.051 for?  Are you just trying to check to see if it is near zero?  Does this value correspond to a certain angle?